### PR TITLE
spice: Fix compilation without deprecated OpenSSL APIs

### PR DESCRIPTION
**Pull request by neheb**
Signed-off-by: Rosen Penev <rosenp@gmail.com>

Maintainer: @yousong 
Compile tested: ath79

Fixes: #10451

### CODE REVIEW

**Comment by neheb** [comment]
Whoops. Updated based on upstream feedback.

File: libs/spice/patches/010-openssl-deprecated.patch [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+ 
+ static int reds_init_ssl(RedsState *reds)
+ {
+@@ -2784,7 +2787,9 @@ static int reds_init_ssl(RedsState *reds)
+     }
+ 
+     SSL_CTX_set_options(reds->ctx, ssl_options);
++#if OPENSSL_VERSION_NUMBER < 0x1010000FL
+     SSL_CTX_set_ecdh_auto(reds->ctx, 1);
++#endif
```
Comment (yousong) [inline libs/spice/patches/010-openssl-deprecated.patch]: This hunk is not present in the upstream patch.

**Comment by neheb** [comment]
That hunk was fixed upstream but not in the released version.

In other news, this patch was accepted upstream.

**Comment by yousong** [comment]
Applied in c0d5c29a242a0e367f3fc276e80f411965d1518d by using upstream patches.

Thanks for your time.

**Pull request closed without merging**